tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -165,79 +189,127 @@ struct bt_mesh_cfg_cli_mod_pub pub_params = {
+ * with a 100ms interval.
+ */
+static struct bt_mesh_test_gatt gatt_param = {
+#if defined(CONFIG_BT_EXT_ADV)
+	/* (total transmit duration) / (transmit interval) */
+	.transmits = 1500 / 100,
+	.interval = 100,
+#else
+	.transmits = 2000 / 1000,
+	.interval = 1000,
+#endif
+	.service = MESH_SERVICE_PROVISIONING,
+};
+
+static bool gatt_check_rx_count(uint8_t transmit)
 {
-	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
-	bt_mesh_device_setup(&prov, &tx_comp);
-	provision_and_configure(tx_cfg, 0);
+	static int cnt;
 
-	k_sem_init(&publish_sem, 0, 1);
-	suspended = false;
-	uint8_t status;
-	int err;
+	LOG_DBG("rx: cnt(%d)", cnt);
+	cnt++;
 
-	err = bt_mesh_cfg_cli_mod_pub_set(0, tx_cfg.addr, tx_cfg.addr, TEST_MODEL_ID_1, &pub_params,
-					  &status);
-	if (err || status) {
-		FAIL("Mod pub set failed (err %d, status %u)", err, status);
+	if (cnt >= transmit) {
+		cnt = 0;
+		return true;
 	}
 
-	/* Wait until node has published before suspending. */
-	for (int i = 0; i < NUM_PUB; i++) {
-		ASSERT_OK_MSG(k_sem_take(&publish_sem, K_SECONDS(30)), "Pub timed out");
+	return false;
+}
+
+static void gatt_scan_cb(const bt_addr_le_t *addr, int8_t rssi,
+			  uint8_t adv_type, struct net_buf_simple *buf)
+{
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_IND) {
+		return;
 	}
 
-	ASSERT_OK(bt_mesh_suspend());
-	suspended = true;
-	LOG_INF("Mesh suspended.");
+	/* Ensure that no message is received while Mesh is suspended or disabled. */
+	ASSERT_FALSE_MSG(dut_status == DUT_SUSPENDED, "Received adv while Mesh is suspended.");
 
-	k_sleep(K_SECONDS(SUSPEND_DURATION));
+	bt_mesh_test_parse_mesh_gatt_preamble(buf);
 
-	ASSERT_OK(bt_mesh_resume());
-	suspended = false;
-	LOG_INF("Mesh resumed.");
+	if (gatt_param.service == MESH_SERVICE_PROVISIONING) {
+		LOG_DBG("Parsing pb_gatt adv");
+		bt_mesh_test_parse_mesh_pb_gatt_service(buf);
+	} else {
+		LOG_DBG("Parsing proxy adv");
+		bt_mesh_test_parse_mesh_proxy_service(buf);
+	}
 
-	for (int i = 0; i < NUM_PUB; i++) {
-		ASSERT_OK_MSG(k_sem_take(&publish_sem, K_SECONDS(30)), "Pub timed out");
+	if (gatt_check_rx_count(gatt_param.transmits)) {
+		LOG_DBG("rx completed, stopping scan...");
+		k_sem_give(&publish_sem);
 	}
+}
 
-	/* Allow publishing to finish before suspending. */
-	k_sleep(K_MSEC(100));
-	ASSERT_OK(bt_mesh_suspend());
+static void gatt_rx_beacons(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type       = BT_HCI_LE_SCAN_PASSIVE,
+		.options    = BT_LE_SCAN_OPT_NONE,
+		.interval   = BT_MESH_ADV_SCAN_UNIT(1000),
+		.window     = BT_MESH_ADV_SCAN_UNIT(1000)
+	};
+	int err;
 
-	PASS();
+	err = bt_le_scan_start(&scan_param, gatt_scan_cb);
+	ASSERT_FALSE_MSG(err && err != -EALREADY, "Starting scan failed (err %d)\n", err);
+
+	err = k_sem_take(&publish_sem, K_SECONDS(10));
+	if (err && (dut_status == DUT_RUNNING)) {
+		FAIL("Tester timed out");
+	} else if ((err == -EAGAIN) && (dut_status == DUT_SUSPENDED)) {
+		LOG_WRN("Tester timed out (DUT is suspended)");
```

Comment:
I see this trigger on all GATT tests. Does it then qualify as a warning from the test POW if it is default behavior?

Reply:
Yeah no it doesn't need a warning. I moved the `EAGAIN` check to `test_tester_gatt` because of the generalized sync/message passing method. 

However... the generalized `bt_mesh_test_wait_for_adv` function will give a warning if waiting on the semaphore times out. This is because the function is used in multiple places (adv and beacon tests).